storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java

```diff
@@ -178,6 +178,43 @@ private static void putIfMissing(Map<String, Double> dest, String destKey, Map<S
         return ret;
     }
 
+    /*
+     * return map with non generic resources removded
+     */
+    public static void filterGenericResources(Map<String, Double> map) {
```

Comment:
can we rename this removeNonGenericResources() or filterOutNonGenericResources() or something?  I think that would be clearer.

Reply:
renamed